Tweak: Add reverse thruster outfits to stock human ships where appropriate.

Hecter94 opened the pull request:
## Summary
After the merging of #5792 there are now two additional reverse thrusters available for purchase in human space. 
However to date, not a single ship comes equipped with either of these two new reverse thrusters and only one ship comes equipped with the previous atomic reverse thruster as part of their stock loadout meaning you rarely ever see them.

The intent of this PR is to increase the prevalence of these new outfits by selectively adding them to several human stock loadouts where appropriate to increase the diversity of builds available to a new player, as well as make the entirety of human space appear more "alive" as ships will now maneuver in different ways based on the thrusters they have installed.

You will now see a lot more ships in human space using reverse thrusters and will have more opportunities to buy ships that come with reverse thrusters. 

See below for a full list of changed ships and my justification for them:
<details>
<summary>Argosy</summary>

- Added Capybara Reverse Thruster

Straight Upgrade. As this ship's stock configuration comes with relatively low turning for its class (42.29 when fully loaded), the addition of reverse thrusters helps it to jump faster, land faster and makes it overall more effective in its role of carrying cargo while still being a capable warship. 
(Modified Argosy removes this reverse thruster to install more weaponry which I feel is fitting) 
With reverse thrusters installed, this ship will use them for both landing and jumping.
</details>


<details>
<summary>Behemoth</summary>

- Added Two X1100 Ion Reverse Thrusters

Straight upgrade, faster landing, slightly faster jumping, overall more efficient in its role
</details>

<details>
<summary>Bulk Freighter</summary>

- Added X1100 Ion Reverse Thruster

Straight upgrade. Better maneuverability, faster landing, faster hyperspace jumps. If we don't feel that a single reverse thruster is enough, we could also downgrade the sidewinders to meteors and install a second reverse thruster.
</details>



<details>
<summary>Container Transport</summary>

- Added X1100 Ion Reverse Thruster

Straight Upgrade. Better maneuverability, faster landing, faster hyperspace jumps, overall improvement to the ship as a whole.
</details>


<details>
<summary>Flivver</summary>

- Downgraded LP036a Battery Pack to Supercapacitor
- Added three KP-6 Photovoltaic Panels
- Added X1100 Ion Reverse Thruster

Added based on Feedback: [1096204460](https://github.com/endless-sky/endless-sky/pull/6710#issuecomment-1096204460)

</details>

<details>
<summary>Freighter</summary>

- Downgraded nGVF-EE Fuel Cell to nGVF-DD Fuel Cell
- Upgraded LP036a Battery Pack to LP072a Battery Pack
- Added X1100 Ion Reverse Thruster

The freighter, as a cargo-carrying vessel will need to jump and land often as part of it's normal operating procedure, both of these tasks will be made more efficient with the addition of reverse thrusters. The decreased energy generation from the downgraded Fuel Cell is also not an issue for this ship as it already had too much energy generation, 354 total generation with a maximum draw of 330 meaning it was running a surplus of 24 energy plus had energy storage on top. 
With this change the ship now performs better with the only expense being that it will need to make use of its upgraded battery in rare cases.
</details>

<details>
<summary>Hauler I</summary>

- Added "Capybara Reverse Thruster"

Straight Upgrade. Helps with moving, jumping, landing and escaping from pirates. Overall increases performance of ship at no cost.
</details>
<details>
<summary>Hauler II</summary>

- Downgraded 2 Meteor Missile Launchers to Missile Pods
- Added 3 Meteor Missile Boxes
- Added "Capybara Reverse Thruster"

Almost a straight upgrade, the two meteor missile pods are slightly less effective than the two launchers, but the total missile supply was kept almost unchanged thanks to the addition of three meteor missile boxes. However the introduction of the reverse thruster helps massively with the overall performance of the ship, increasing the speed of jumps, landing and makes escaping from combat a bit easier.
</details>
<details>
<summary>Hauler III</summary>

- Downgraded 2 Meteor Missile Launchers to Missile Pods
- Added 3 Meteor Missile Boxes
- Added "Capybara Reverse Thruster"

Almost a straight upgrade, the two meteor missile pods are slightly less effective than the two launchers, but the total missile supply was kept almost unchanged thanks to the addition of three meteor missile boxes. However the introduction of the reverse thruster helps massively with the overall performance of the ship, increasing the speed of jumps, landing and makes escaping from combat a bit easier.
</details>
<details>
<summary>Skein</summary>

- Downgraded Meteor Missile Launchers to Meteor Missile Pods
- Added a Meteor Missile Box
- Removed Large Radar Jammer
- Added Capybara Reverse Thruster

Added based on feedback in: https://github.com/endless-sky/endless-sky/pull/6710#issuecomment-1098653969
Skein as one of the largest FW warships is remarkably slow to maneuver, especially considering it's role as a carrier. Adding a reverse thruster greatly increases it's jump speed, landing speed, and combat maneuverability. The change to the meteor missile pods does reduce it's overall ammo capacity, however with the addition of the meteor missile box, the total capacity of meteor missiles only drops by 31.
Additionally, while the original Hauler design is designed for simplicity and thus a reverse thruster wouldn't be appropriate, the Skein is designed as a warship and a little extra complexity from reverse thrust is justifiable on such a large and important part of the FW fleet.
</details>


Notes:

- As these are simply tweaks to stock loadouts and not changes to the hull of the ships, this does not impact the limits of the player. If they wish to remove the reverse thrusters and add back weapons/reactors/cooling, they are free to do so.
- This PR does **not** add reverse engine flares, this is by intent. These outfits can be swapped between ships at will, and as such it does not make sense for one ship to have flares whereas the same outfit on a different ship doesn't have flares.
- I would suggest a sweeping change to all human ships to give all of them reverse flares at the same time in order to avoid this inconsistency. That would, however be a change outside the scope of this PR.
- Likewise, this PR is not currently editing Marauders or Variants. However if these changes are accepted, I will handle any Marauder and Variant changes in a separate PR

## Testing
I have confirmed that every one of these ships is displaying with the edited loadout in the appropriate shipyards
I have tested each ship to confirm that they are able to reverse as expected and that they still function in their intended role after the stock loadout changes. 

## Save File
This save file will place you on New Boston with every edited ship already in your fleet:
[Reverser.Test.Save.txt](https://github.com/endless-sky/endless-sky/files/8492179/Reverser.Test.Save.2.txt)




## PR Checklist
~~[X] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified~~
~~[X] I uploaded the necessary image, blend, and texture assets here: {{insert link to assets}}~~
~~[X] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the @2x versions of these art assets: {{insert PR link}}~~

Anarchist2 commented:
Some notes on how much the AI uses reverse in practice:
* The Argosy uses reverse thrust for landing and jumping.
* The Falcon and Splinter use reverse thrust for landing and jumps that have a shallow angle (they use reverse on the Graffias-Alniyat-Dschubba jump, but not on the Albireo-Lurata-Dabih route).
* The Arrow, Bounder, Corvette, Osprey and Raven will very rarely use their reverse thrusters, only using them on the first jump after a take off.
* The (Bulk) Freighter will only use reverse thrusters on landing, preferring to use their main engines with the Scram Drive.
* Reverse thrusters were some use when attempting to jump from a system while being hit in small vessels, but had little improvement when used on large vessels (Tested with an Argosy, a Bounder and a Freighter).
* No comment on how the AI uses reverse thrust during combat.

I'd also much prefer if the Raven was dropped a shield generator size instead of mounting a more expensive generator.

Hecter94 commented:
> Some comments on how much the AI uses reverse in practice:
> 
> * The Argosy uses reverse thrust for landing and jumping.
> * The Berserker, Falcon and Splinter use reverse thrust for landing and jumps that have a shallow angle.
> * The Arrow, Bounder, Corvette, Osprey and Raven will very rarely use their reverse thrusters, only using them on the first jump after a take off.
> * The (Bulk) Freighter will only use reverse thrusters on landing, preferring to use their main engines with the Scram Drive.
> * Reverse thrusters were of some use when attempting to jump from a system while being hit in small vessels, but had little improvement when used on large vessels (Tested with an Argosy, a Bounder and a Freighter).
> * No comment on how the AI uses reverse thrust during combat.
> 
> I'd also much prefer if the Raven was dropped a shield generator size instead of mounting a more expensive generator.

Thank you for testing! That is very helpful information to have.

For the Arrow, Bounder, Corvette, Osprey and Raven, do you feel there would be value in further tweaking the build to upgrade their reverse thrusters or install an additional one? So that they make use of their reverse thrust more? Or leave them as-is?

Looking at it, the Arrow would be difficult to upgrade to the next tier of reverse thrust without making a hull change or compromising its build. However the Bounder, Corvette, Osprey and Raven could be upgraded to at least a Capybara quite easily.

I also agree with your assessment of the Raven, dropping the shield generator only loses 11.4 shield regen, but drops the cost of the ship down to 3.107m from 3.637m. I feel that is a worthwhile tradeoff so I have made this change.

Hecter94 commented:
> The Bounder can't mount a larger reverse thruster; it's too short on weapon space. Honestly, I don't feel like the Bounder really benefits at all from reverse thrusters; its turning rate is already high enough to outmaneuver most everything that comes its way. 

I would like to keep reverse thrusters on the bounder if possible, it's description states that it was designed for carrying cargo and passengers across large distances quickly, something that would benefit greatly from reverse thrust, making most hyperjumps and landing faster overall.

I was thinking of replacing it's two anti-missile turrets with a single heavy anti-missile turret and then upgrading the Ion Reverse Thruster to a Capybara Reverse Thruster. That would use up all of its weapon space, with 4 outfit space free.

Based on this AM chart I made, a single Heavy Anti-Missile is roughly comparable or better compared to two Light Anti-Missiles in most categories so it would not be a downgrade, and the improved reverse thruster would make it use it more often.
https://docs.google.com/spreadsheets/d/1AVC8737FPOd3arzpdZq3RnHd8rrYLlBzqaMN7ha-nis/edit#gid=1694471554

With the Capybara installed it makes extensive use of it's reverse thrust, only preferring to use steering on very sharp turns, such as Vega -> Sol -> Caph.

> I'd instead put reverse thrusters on something like the Flivver, swapping the battery out for 3 panels and a supercapacitor. Its low turn speed means that it's greatly improved by reverse thrusters, and the comfort of not needing to go through high-degree turns was the reason reverse thrusters were put on the Star Queen (Changing the Flivver's loadout brings it up to costing 522 thousand credits, a 10% increase in its cost... but only 3000 more credits than the 0.9.14 Flivver). The Arrow would probably also qualify under similar luxury reasoning, but as you've noted, it can't fit an engine large enough that the AI uses it.

I can see the Flivver qualifying for reverse thrust under the same "luxury" line of reasoning as the Star Queen and the Arrow, so I have gone ahead and edited the stock loadout to include one, and with testing, it does seem to benefit far more from the reverse thrust than the Arrow does. In addition to that, the ship's description states that one of it's only selling point is speed and if you take "speed" to include hyperjumps and landing, that selling point is greatly improved with the addition of reverse thrust. I think this makes sense both in-lore and mechanically. 
So I accept this change, and have reverted the Arrow back to its stock config and have given the Flivver a reverse thruster instead.

> As for the other ships, upgrading their reverse thrusters makes them jump and land with reverse thrusters. However, this causes the Osprey to use atomic thrusters, despite being produced an entire galaxy away from them. While this isn't a be-all end-all problem (see Heavy Anti-missile Turrets), atomic engines are generally tightly restricted to a select few ships, with those ships being described as especially opulent.

Given that the Osprey is also described as being "Overpriced" and an "Agile light warship" for those "who have the extra cash to spend on it". I feel that it is more than opulent enough to justify mounting a single small atomic engine. Engines that are by their own definition designed for captains who are willing to pay a bit extra for extra speed and agility.

> Is it intentional that the Hauler line(s) aren't touched in this PR? Seeing as they lack scram drives, they would benefit far more from reverse thrust than the Syndicate Freighters.

Yes, this is intentional, based on the description of the Hauler stating "Haulers were first created simply by bolting a cockpit and engine block onto a cargo container, and the basic design has not changed much since then." I feel as though Haulers were designed to be a very simplistic ship with a very low manufacturing cost, and adding reverse thrusters felt like overcomplicating a design that is supposed to be very simple.
I agree that they would probably benefit more from having reverse thrusters, but I don't feel like that matches with what the Hauler line is designed to be, so I would like to leave that for the player to experiment with.

AvianGeneticist commented:
The Class-C Freighter having a weaker stock armament compared to the bulk freighter doesn't make much sense to me; at least not for the reasoning you used. It's not a bait ship in the sense that it's weak by itself; it's bait in the sense of Batesian Mimicry, where it looks similar to a more/less dangerous object in order to more effectively exist. Changing the stock armament actively shows off "hey, this is a different ship model, learn the differences, dingnuts".

I do want to contribute positively here, so:
The Skein should have reverse thrust. It's too slow and cumbersome for their basic engine blocks to really make it motile, even with judicious use of atomics. Being military vessels, any "they're supposed to be basic and cheap" conditions are easily handwaved. When your security is on the line, you bring out the good stuff. I do agree that the basic haulers don't need reverse thrust, though.

Replacing the Manta's missile launchers with a reverse thruster could better reinforce its role as a picket ship (and an upgrade over the Quicksilver).

Removing the Vanguard's anti-missile turret and replacing it with a reverse thruster could also prove nice. The vanguard, with its insane emphasis on frontal firepower, relies more on good positioning than any other heavy warship, so giving it the ability to course-correct to "joust" less against ships with more agile turrets would be a welcome change.
(The only issue I see here is that core syndicate ships tend to be as bare-bones as they come. That said, maybe the removal of the turret is justification enough?)

Hecter94 commented:
Thanks for the reply! I appreciate it!

> The Class-C Freighter having a weaker stock armament compared to the bulk freighter doesn't make much sense to me; at least not for the reasoning you used. It's not a bait ship in the sense that it's weak by itself; it's bait in the sense of Batesian Mimicry, where it looks similar to a more/less dangerous object in order to more effectively exist. Changing the stock armament actively shows off "hey, this is a different ship model, learn the differences, dingnuts".

I completely agree with this, and it does make a lot of sense. 
With that said, the Bulk Freighter actually _does_ come with Laser Turrets as it's stock loadout. The Class C Freighter upgrades those to Heavy Laser Turrets for its stock loadout.
This PR, by dropping the Heavy Laser Turrets on the Class-C Freighter down to regular laser turrets actually makes it match the Bulk Freighter _better_

By default the Bulk Freighter comes with 3 Laser Turrets and 2 Heavy Anti-Missile Turrets, and the Class-C Freighter comes with 2 Heavy Laser Turrets and 2 Heavy Anti-Missile Turrets. 
With this PR, the Class-C Freighter comes with 2 Laser Turrets and 2 Heavy Anti-Missile Turrets, which is about as close as it can come to matching the Bulk Freighter without tweaking the Bulk Freighter's loadout. 

> I do want to contribute positively here, so: The Skein should have reverse thrust. It's too slow and cumbersome for their basic engine blocks to really make it motile, even with judicious use of atomics. Being military vessels, any "they're supposed to be basic and cheap" conditions are easily handwaved. When your security is on the line, you bring out the good stuff. I do agree that the basic haulers don't need reverse thrust, though.

This makes total sense as well, and resolves the concern I would have had regarding cost and complexity. By "Skein", would you like the entire line of Roost/Nest/Skein be given reverse thrust? Or just the Skein? This is something I could definitely look into adding though. 

> Replacing the Manta's missile launchers with a reverse thruster could better reinforce its role as a picket ship (and an upgrade over the Quicksilver).

I'm okay with this change, removing the two Meteor Missile Launchers as well as the Meteor Missile Box gives enough room to install a Capybara reverse thruster and leaves 9 outfit space left over for expansion. 
The only real issue is that the AI won't actually make much use of this reverse thrust in practice, but it is still useful for the player and still useful for general mobility.
I'll go ahead and add this into the PR as I do think it makes sense. 

> Removing the Vanguard's anti-missile turret and replacing it with a reverse thruster could also prove nice. The vanguard, with its insane emphasis on frontal firepower, relies more on good positioning than any other heavy warship, so giving it the ability to course-correct to "joust" less against ships with more agile turrets would be a welcome change. (The only issue I see here is that core syndicate ships tend to be as bare-bones as they come. That said, maybe the removal of the turret is justification enough?)

I'm not sure about this though, given how prevalent missiles are in human space, I just can't see the syndicate willingly removing the only anti-missile that one of their ships has. If I could somehow fit a reverse thruster without removing its anti-missile and without downgrading its weapons I'd be all for it, but I just can't justify removing the Vanguard's only anti-missile even if it would make the ship more effective at dealing damage with it's mounted guns.

AvianGeneticist commented:
> I completely agree with this, and it does make a lot of sense. With that said, the Bulk Freighter actually _does_ come with Laser Turrets as it's stock loadout. The Class C Freighter upgrades those to Heavy Laser Turrets for its stock loadout. This PR, by dropping the Heavy Laser Turrets on the Class-C Freighter down to regular laser turrets actually makes it match the Bulk Freighter _better_

> By default the Bulk Freighter comes with 3 Laser Turrets and 2 Heavy Anti-Missile Turrets, and the Class-C Freighter comes with 2 Heavy Laser Turrets and 2 Heavy Anti-Missile Turrets. With this PR, the Class-C Freighter comes with 2 Laser Turrets and 2 Heavy Anti-Missile Turrets, which is about as close as it can come to matching the Bulk Freighter without tweaking the Bulk Freighter's loadout.

Did not know there was such a difference, interesting. (Begs the difference as to why the Syndis removed the turret mount at all.)


> This makes total sense as well, and resolves the concern I would have had regarding cost and complexity. By "Skein", would you like the entire line of Roost/Nest/Skein be given reverse thrust? Or just the Skein? This is something I could definitely look into adding though.

I was thinking just the Skein, but it could be extended to the full line for consistency. The Nest and Roost are generally okay on their own with some fanagling, and iirc the Nest actually moves at fair speeds with no customization.

> I'm not sure about this though, given how prevalent missiles are in human space, I just can't see the syndicate willingly removing the only anti-missile that one of their ships has. If I could somehow fit a reverse thruster without removing its anti-missile and without downgrading its weapons I'd be all for it, but I just can't justify removing the Vanguard's only anti-missile even if it would make the ship more effective at dealing damage with it's mounted guns.

This is a fair point, and sadly there's no real way around it. Even the smallest reverse thruster is too big to squeeze into a Vanguard's (rather paltry) weapon space, all other aspects of its kit left withsanding.
If not stock, perhaps certain variants of the Vanguard could be given reverse thrust instead? I'm sure there are a number of pilots who'd prefer some more offense control to defense.
I know you aren't messing with variants here much, but players who focus on capping might fight some intrigue in new toys.

Hecter94 commented:
> I was thinking just the Skein, but it could be extended to the full line for consistency. The Nest and Roost are generally okay on their own with some fanagling, and iirc the Nest actually moves at fair speeds with no customization.

This makes total sense! I think I will just go with the Skein for now as I want to limit reverse thrust to only a handful of ships, and the Skein needs it the most. So I will go ahead and add that to this PR as well.

> This is a fair point, and sadly there's no real way around it. Even the smallest reverse thruster is too big to squeeze into a Vanguard's (rather paltry) weapon space, all other aspects of its kit left withsanding. If not stock, perhaps certain variants of the Vanguard could be given reverse thrust instead? I'm sure there are a number of pilots who'd prefer some more offense control to defense. I know you aren't messing with variants here much, but players who focus on capping might fight some intrigue in new toys.

Yeah, definitely! If this PR ends up getting merged, then when I take a look at Variants I'll see if there's anything I can do with the Vanguard, since it would be cool to have a variant with a bit of reverse thrust.

Amazinite reviewed (changes requested):
Insofar as the AI doesn't actually effectively utilize reverse thrusters in combat, I don't think we should be adding reverse thrusters to the stock loadouts on combat ships. That is to say, I'm not really interested in the "lore" claims about the way certain combat ships are suppose to fight and how they'd utilize their reverse thrusters when they don't actually do that and think that we should remove the changes made to the Berserker, Falcon, Manta, Osprey, Raven, and Splinter. (Whenever AI changes are made to the use of reverse thrusters in combat then we could revisit this.) The freighter and transport changes can stay since they'll at least (theoretically) put reverse thrust to use when jumping and landing, and such ships rely on being able to quickly jump between systems and land. The Skein changes can stay since it's a modified freighter, so the reverse thrusters on a (somewhat) combat ship can be said to have come from that lineage. The Class C Freighter isn't even available in the game yet, so revert the changes made to it.

Hecter94 commented:
> we should remove the changes made to the Berserker, Falcon, Manta, Osprey, Raven, and Splinter. The Skein changes can stay since it's a modified freighter, so the reverse thrusters on a (somewhat) combat ship can be said to have come from that lineage. The Class C Freighter isn't even available in the game yet, so revert the changes made to it.

Understood, I have removed the changes to the Berserker, Falcon, Manta, Osprey, Raven, Splinter and Class-C. I will revisit these once the Class C is made available and when the AI changes to reverse thrust happen. 
Considering we're saying that the reverse thrusters on the Skein came from it's hauler lineage, would you like me to go ahead and add reverse thrusters to the Hauler line as well?

Amazinite commented:
> would you like me to go ahead and add reverse thrusters to the Hauler line as well?

Go ahead.

Hecter94 commented:
I have added reverse thrusters to the entire Hauler line, I also removed reverse thrusters from the Corvette as it seems to me like that falls under the definition of "combat ship" which I will be revisiting only once the AI is improved to handle reverse thrust in combat better. 
Some notes on how the AI of each ship uses their thrusters after changes since Anarachist2s tests:

**Uses reverse thrust for landing and jumping**
- Argosy
- Hauler 
- Hauler II
- Hauler III
- Skein
- Flivver

**Uses reverse thrust for landing, prefers main thruster when jumping**
- Bulk Freighter
- Container Transport
- Freighter

**Rarely uses its reverse thruster except when first launching and very shallow turns**
- Bounder

Based on these results, I am thinking of reverting the bounder to its stock config, and giving reverse thrusters to the Behemoth instead. It has more than enough space to fit one and with testing, makes much better use of it.
